ament_uncrustify/ament_uncrustify/main.py

```diff
@@ -301,12 +226,118 @@ def get_files(paths, extensions, excludes=[]):
+    cwd = None
+    if os.name == 'nt':
+        cwd = os.path.commonprefix(input_files)
+        if not os.path.isdir(cwd):
+            cwd = os.path.dirname(cwd)
+            assert os.path.isdir(cwd), \
+                'Could not determine common prefix of input files'
+        input_files = [os.path.relpath(f, start=cwd) for f in input_files]
+
+    try:
+        cmd = [
+            uncrustify_bin,
+            '-c', args.config_file,
+            '-l', language,
+            '--prefix', temp_path,
+            '--suffix', suffix]
+        cmd.extend(input_files)
+        subprocess.check_output(cmd, cwd=cwd, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        if e.output:
+            print(e.output.decode(), file=sys.stderr)
+        print("The invocation of 'uncrustify' failed with error code %d: %s" %
+            (e.returncode, e), file=sys.stderr)
+        return 1
+
+    if cwd:
+        # input files are relative
+        # prepend temp path, append suffix
+        output_files = [
+            os.path.join(temp_path, f + suffix) for f in input_files]
+    else:
+        # input files are absolute
+        # remove leading slash, prepend temp path, append suffix
+        output_files = [
+            os.path.join(
+                temp_path,
+                os.sep.join(f.split(os.sep)[1:]) +
+                suffix
+            ) for f in input_files
+        ]
+
+    uncrustified_files = output_files
+    i = 1
+    while True:
+        # identify files which have changed since the latest uncrustify run
+        changed_files = []
+        for input_filename, output_filename in zip(
+                input_files, uncrustified_files):
+            if cwd and not os.path.isabs(input_filename):
+                input_filename = os.path.join(cwd, input_filename)
+            if not filecmp.cmp(input_filename, output_filename):
+                if output_filename == input_filename + suffix:
+                    # for repeated invocations
+                    # replace the previous uncrustified file
+                    dst_filename = input_filename
+                else:
+                    # after first invocation remove suffix
+                    # otherwise uncrustify behaves different
+                    output_filename_without_suffix = \
+                        output_filename[:-len(suffix)]
+                    dst_filename = output_filename_without_suffix
+                if os.path.exists(dst_filename):
+                    # on Windows os.rename doesn't replace an existing dst
+                    os.remove(dst_filename)
+                os.rename(output_filename, dst_filename)
+                changed_files.append(dst_filename)
+        if not changed_files:
+            break
+        # reinvoke uncrustify for previously changed files
+        input_files = changed_files
+        try:
+            cmd = [uncrustify_bin, '-c', args.config_file, '--suffix', suffix]
```

Comment:
@dirk-thomas uh, isn't the language flag necessary here as well?

Reply:
Absolutely, good catch: b3f4a448c19444ae98103bfba25e1aa7470ed93c

And since language can be `None` it needs to be added conditionally.